lib/src/notification/notification.component.ts

```diff
@@ -0,0 +1,176 @@
+    }
+
+    /**
+     * Get the theme class based on the theme passed though the props
+     */
+    private getThemeClass(): void {
+        let themeClass: string = "theme-";
+        if (
+            this.theme &&
+            ["purple", "primary", "danger", "success", "warning", "inverted"].some((t: NotificationTheme) => t === this.theme)
+        ) {
+            themeClass += this.theme;
+        } else {
+            themeClass += "purple";
+        }
+        this.notificationClassNames += " " + themeClass;
+    }
+
+    /**
+     * Get the position class based on the position and style passed through the props
+     */
+    private getPositionClass(): void {
+        let positionClass: string;
+        const position: string = this.position;
+        if (this.style && ["slide-in", "bar"].some((s: string) => s === this.style)) {
+            switch (this.style) {
+                case "slide-in":
+                    if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].some((p: string) => p === position)) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "bottom-left";
+                    }
+                    break;
+                case "bar":
+                    if (position && ["top", "bottom"].some((p: string) => p === position)) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "top";
+                    }
+                    break;
+            }
+        } else {
+            // Should default back to `slide-in`
+            if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].some((p: string) => p === position)) {
+                positionClass = position;
+            } else {
+                positionClass = "bottom-left";
+            }
+        }
+
+        this.notificationClassNames += " " + positionClass;
+    }
+
+    /** Start the timer to dismiss the notification */
+    private startTimer(): void {
+        this.clearTimer();
+        this.timerRef = window.setTimeout(() => {
+            this.dismiss();
+        }, this.dismissTimeout || this.defaultTimeout);
+    }
+
+    /** Dismiss the notification */
+    private dismiss(): void {
+        this.clearTimer();
+        this.onDismiss && this.onDismiss.emit();
+    }
+
+    /** Clear the timer that dismisses the notification */
+    private clearTimer(): void {
+        if (this.timerRef) {
+            clearTimeout(this.timerRef);
+            this.timerRef = null;
+        }
+    }
+
+    private setClassNames(): void {
+        this.notificationClassNames = "custom-notification";
+        this.getStyleClass();
+        this.getThemeClass();
+        this.getPositionClass();
+
+        this.notificationClassNames += this.toggle ? " open" : "";
```

Comment:
This is not the way of constructing a class object in angular use this directive https://angular.io/api/common/NgClass